internal/handlers/pg/pgdb/query.go

```diff
@@ -246,20 +246,38 @@ func prepareWhereClause(sqlFilters *types.Document) (string, []any) {
 	var p Placeholder
 
 	for k, v := range sqlFilters.Map() {
-		switch k {
-		case "_id":
-			switch v := v.(type) {
-			case string:
-				filters = append(filters, fmt.Sprintf(`((_jsonb->'_id')::jsonb = %s)`, p.Next()))
-				args = append(args, string(must.NotFail(pjson.MarshalSingleValue(v))))
+		// don't pushdown query operators and dot notation yet
+		if len(k) != 0 && (k[0] == '$' || types.NewPathFromString(k).Len() > 1) {
+			continue
+		}
 
-			case types.ObjectID:
+		// don't iterate through array for _id keys to simplify the query
+		if k == "_id" {
+			switch v := v.(type) {
+			case types.ObjectID, string:
 				filters = append(filters, fmt.Sprintf(`((_jsonb->'_id')::jsonb = %s)`, p.Next()))
 				args = append(args, string(must.NotFail(pjson.MarshalSingleValue(v))))
 			}
-		default:
+
 			continue
 		}
+
+		switch v := v.(type) {
+		case string, int32, int64, types.ObjectID, float32, float64:
+			sql := fmt.Sprintf(
+				`((_jsonb->%[1]s)::jsonb = %[2]s)`+ // Select if value under the key k is equal to value v.
+					// If it's not, but the value under the key k is an array - select if it contains the value equal to v.
+					` OR (jsonb_typeof(_jsonb->%[1]s) = 'array' AND`+
+					` EXISTS (SELECT 1 FROM jsonb_array_elements(_jsonb->%[1]s) as elem WHERE (elem::jsonb = %[2]s)))`,
+				p.Next(), // $1 = k
+				p.Next(), // $2 = v
```

Comment:
Can we escape or sanitize before using p.Next()?

Reply:
Where? We don't push data directly to the query, but use the arguments. We've also used the same approach before this PR. I guess I don't understand

Reply:
`p.Next()` must be escaped or sanitized, so it does not contain harmful query.

>  We've also used the same approach before this PR. I guess I don't understand
Please fix them as well please :pray:

Reply:
I see, I misunderstood initially. it's value is `$2`, not `v` which I thought was a value of something :see_no_evil:.